src/Festung/Frontend/Converters.hs

```diff
@@ -31,21 +31,33 @@ vaultObjectParser :: Value -> Parser P.Value
 vaultObjectParser (String s) = return $ P.StringValue (T.unpack s)
 vaultObjectParser (Number n) = return $ either P.FloatValue P.IntValue $ floatingOrInteger n
 vaultObjectParser Null       = return P.NullValue
-vaultObjectParser _          = fail "Can't convert to a SQLCipher type"
+vaultObjectParser any          = fail ("Can't convert to a SQLCipher type: " ++ show any)
 
 
-parametersParser :: Value -> Parser [P.Value]
-parametersParser (Array a)  = mapM vaultObjectParser (V.toList a)
-parametersParser (Object _) = fail "Not implemented yet"
-parametersParser _          = fail "Parameters must be an array, or a mapping of parameters"
+parametersParser :: Value -> Parser P.Row
+parametersParser (Array  a) = mapM vaultObjectParser (V.toList a)
+parametersParser _ = fail "Not implemented yet"
 
 
-queryParser :: Value -> Parser (String, [P.Value])
-queryParser = withObject "Should be a query object" $ \ obj -> do
-    query       <- obj .: "sql"
-    paramsArray <- obj .:? "params" .!= emptyArray
-    params      <- parametersParser paramsArray
-    return (query, params)
+bulkParametersParser :: Value -> Parser [P.Row]
+bulkParametersParser (Array a) = mapM parametersParser (V.toList a)
+bulkParametersParser _ = fail "Not implemented yet"
+
+
+chooseParams :: P.Row -> [P.Row] -> P.QueryParams
+chooseParams [] [] = P.Params []
+chooseParams [] ps = P.BulkParams ps
+chooseParams ps _  = P.Params ps
+
+
+queryParser :: Value -> Parser (String, P.QueryParams)
+queryParser = withObject "Should be a query object" $ \obj -> do
+    query           <- obj .: "sql"
+    paramsArray     <- obj .:? "params" .!= emptyArray
+    params          <- parametersParser paramsArray
+    bulkParamsArray <- obj .:? "bulkParams" .!= emptyArray
```

Comment:
I would prefer if the field in the request body would be in snake-case (e.g.`"bulk_params"`), because that is more inline with the format of the fields in the response (e.g. `"last_row_id"`).